tracing-subscriber/src/registry/sharded.rs

```diff
@@ -89,6 +89,24 @@ fn id_to_idx(id: &Id) -> usize {
     id.into_u64() as usize - 1
 }
 
+pub(crate) struct LayerGuard<'a> {
+    id: Id,
+    registry: &'a Registry,
+}
+
+impl<'a> Drop for LayerGuard<'a> {
+    fn drop(&mut self) {
+        CLOSE_COUNT.with(|count| {
```

Comment:
`LocalKey::with` [will panic](https://doc.rust-lang.org/std/thread/struct.LocalKey.html#panics) if the thread local has already been destroyed. Since this is a `Drop` impl, it can be called while unwinding. 

We should avoid a double panic by calling [`LocalKey::try_with`](https://doc.rust-lang.org/std/thread/struct.LocalKey.html#method.try_with) instead. If the `try_with` fails, we are panicking, so I think okay to do nothing in that case and allow the span to be "leaked", since the panic will nuke the span storage anyway.

Reply:
fixed, thanks!